maro/rl/rollout/env_sampler.py

```diff
@@ -253,25 +265,27 @@ def rl_policy_dict(self) -> Dict[str, RLPolicy]:
     @abstractmethod
     def _get_global_and_agent_state(
         self, event: object, tick: int = None,
-    ) -> Tuple[Optional[np.ndarray], Dict[Any, np.ndarray]]:
+    ) -> Tuple[Union[None, np.ndarray, List[object]], Dict[Any, Union[np.ndarray, List[object]]]]:
         """Get the global and individual agents' states.
```

Comment:
also add comment like 
```
If the policy is a `RLPolicy`, its state is a Numpy array. Otherwise, its state is a list of objects.
```

here? Since it is the interface that should be implemented by users.